### PR TITLE
Add Django 5 and python 3.12 and 3.13 test support

### PR DESCRIPTION
**Pull request by dancergraham**
based on https://github.com/jazzband/django-waffle/pull/484/commits

### CODE REVIEW

**Review by ulgens — CHANGES_REQUESTED** [review]
I'm not sure if I'm the right person to give reviews on this package but I think other people will agree that this PR does too much. Python 3.8 removal, Github Actions updates, partial Django 3.2 removal, 4.0 and 4.1 removal, Django 5.1 support and then Python 3.12 and 3.13 support. Almost all of them deserve their own comprehensive PR.

**Comment by clintonb** [comment]
I agree that there is too much here. Please create separate pull requests for updating Actions and adding support for new Python and Django versions. We can drop support for EOL’d versions in a later, major, release.

**Comment by dancergraham** [comment]
Hi thanks for the review ! yes I agree that this is mixing subjects, as did the PR I based it on. I will try to find some time to redo it in separate steps...

File: tox.ini [hunk 1]
```diff
@@ -1,32 +1,32 @@
...
 [testenv]
 allowlist_externals = ./run.sh
 deps =
-	django32: Django>=3.2,<3.3
-	django40: Django>=4.0,<4.1
-	django41: Django>=4.1,<4.2
 	django42: Django>=4.2,<4.3
+    django50: Django>=5.0,<5.1
+	django51: Django>=5.1,<5.2
+
```
Comment (cclauss) [inline tox.ini:21]:
```suggestion
	django50: Django>=5.0,<5.1
	django51: Django>=5.1,<5.2
```

File: pyproject.toml [hunk 2]
```diff
@@ -17,20 +17,22 @@ classifiers = [
     "License :: OSI Approved :: BSD License",
     "Operating System :: OS Independent",
     "Framework :: Django",
-    "Framework :: Django :: 3.2",
-    "Framework :: Django :: 4.0",
-    "Framework :: Django :: 4.1",
+    "Framework :: Django :: 4.2",
+    "Framework :: Django :: 5.0",
+    "Framework :: Django :: 5.1",
+    "Framework :: Django :: 5.2",
```
Comment (cclauss) [inline pyproject.toml:23]:
```suggestion
```

**Review by cclauss — APPROVED** [review]
Nice!  Git conflicts to resolve.

**Comment by dancergraham** [comment]
Handled elsewhere - 9 PRs is better than one ;)

**Pull request closed without merging**